feat(replay) - added playlist tab view

jerryzhou196 opened the pull request.

jerryzhou196 committed:
feat(replays): add playlist tab and component to replay details

jerryzhou196 committed:
Merge branch 'master' into replay-playlist-tab-view

jerryzhou196 committed:
fixed replay table header being resized

jerryzhou196 committed:
ref(replay): Enhance ReplayPlaylistProvider to include currentReplay index

- Updated ReplayPlaylistProvider to accept currentReplay as a prop and calculate its index within the replays array.
- Adjusted usage in ReplayDetailsProviders to pass the current replay.
- Modified ReplayDetailsPageBreadcrumbs to utilize the new structure for accessing replays and currentReplayIndex.

jerryzhou196 committed:
feat(replay): Enhance ReplayTable with highlighted row functionality

- Added highlightedRowIndex prop to ReplayTable for improved user experience.
- Implemented smooth scrolling to the highlighted row when the index changes.
- Updated Playlist component to pass currentReplayIndex and location query to ReplayTable.
- Modified ReplayListQueryReferrer to include 'replaysPlayList' for better query handling.

jerryzhou196 committed:
ref(replay): Update ReplayDetailsPageBreadcrumbs to use initialLocation ref for query handling

- Replaced direct usage of location.query with initialLocation.current.query to maintain clean URL state during navigation.
- Ensured consistent query handling for next and previous replay navigation.

billyvg reviewed (commented):
This is a very focused PR, good job!

Is this feature-flagged?

jerryzhou196 committed:
ref(replay): Refactor ReplayTable to improve row highlighting and scrolling behavior

- Removed the previous useEffect for scrolling to the highlighted row and implemented a new RowWithScrollIntoView component.
- The new component handles scrolling into view when a row is hovered, enhancing user experience.
- Updated the data attribute for row indexing to follow a consistent naming convention.

jerryzhou196 commented on static/app/components/replays/table/replayTable.tsx:
```diff
@@ -40,16 +41,18 @@ export default function ReplayTable({
+  highlightedRowIndex = -1,
   sort,
 }: Props) {
+  const tableRef = useRef<HTMLDivElement | null>(ref?.current ?? null);
```
ref seems unnecessary

jerryzhou196 commented on static/app/views/replays/types.tsx:
```diff
@@ -203,7 +203,8 @@ export type ReplayListLocationQuery = {
   | 'issueReplays'
-  | 'transactionReplays';
+  | 'transactionReplays'
+  | 'replaysPlayList';
```
is this enum used anywhere?

jerryzhou196 committed:
ref(replay): Simplify ReplayTable by removing unnecessary useRef and updating row highlighting logic

- Removed the useRef for table reference, directly using the provided ref instead.
- Updated the row highlighting logic to use scrollIntoView for better clarity and functionality.
- Cleaned up the props in RowWithScrollIntoView for improved readability.

jerryzhou196 committed:
ref(replay): Update ReplayListQueryReferrer to remove 'replaysPlayList' for cleaner query handling

jerryzhou196 committed:
ref(replay): Integrate mobile project support in Playlist component

- Added useAllMobileProj hook to determine if mobile project columns should be displayed.
- Introduced MOBILE_COLUMNS for mobile-specific replay data presentation.
- Updated columns prop in ReplayTable to conditionally render based on mobile project availability.

jerryzhou196 commented on static/app/components/replays/table/replayTable.tsx:
```diff
@@ -121,12 +129,38 @@ export default function ReplayTable({
+function RowWithScrollIntoView({
+  children,
+  data_replay_row_index,
+  scrollIntoView: isHovered,
```
remove unnecessary alias

jerryzhou196 committed:
ref(replay): Update RowWithScrollIntoView to use consistent prop naming

- Changed the prop name from 'isHovered' to 'scrollIntoView' for clarity.
- Updated the useEffect dependency to reflect the new prop name, ensuring correct scrolling behavior when the prop changes.

jerryzhou196 committed:
ref(replay): Conditionally render Playlist tab based on organization features

- Updated the Playlist tab to only display if the organization has the 'replay-playlist-view' feature enabled, improving flexibility in the UI.

jerryzhou196 committed:
Merge branch 'master' into replay-playlist-tab-view

jerryzhou196 committed:
refactor(replayTable): Simplify row scrolling logic and enhance row component with ref forwarding - Removed unnecessary data attribute for row index and replaced it with a ref for scrolling.
- Updated RowWithScrollIntoView to utilize useRef for scrolling into view.
- Enhanced SimpleTable.Row to accept a ref prop for better integration with React's ref system.

jerryzhou196 committed:
refactor(replayDetails): Update replays mapping to return undefined instead of an empty array - Improved clarity in the useMemo hook for replay data processing.

jerryzhou196 committed:
fix(playlist): Wrap ReplayTable in a div so the table doesn't try to fill the entire container

jerryzhou196 commented on static/app/components/replays/table/replayTable.tsx:
```diff
@@ -121,12 +128,33 @@ export default function ReplayTable({
+      {children}
+    </SimpleTable.Row>
+  );
+}
```
code reviewed with Billy - this seems fine to ignore

jerryzhou196 committed:
refactor(playlist): Remove unnecessary div wrapper around ReplayTable for cleaner structure

jerryzhou196 committed:
style(replayTable): Add gridAutoRows to style for improved row height management

jerryzhou196 committed:
refactor(replayPlaylist): Update replay state initialization to use undefined instead of an empty array for improved clarity in context management

jerryzhou196 committed:
refactor(replayTable): Simplify RowWithScrollIntoView by removing unnecessary variant prop from SimpleTable.Row for cleaner component structure, in addition to reordering so rowRef can override

jerryzhou196 committed:
style(replayTable): Remove inline gridAutoRows from StyledSimpleTable and apply it as a separate styled property for improved style management

jerryzhou196 committed:
Merge branch 'master' into replay-playlist-tab-view

jerryzhou196 committed:
refactor(replayPlaylist): Update replay handling to ensure consistent initialization and avoid undefined states

- Changed the `replays` prop in `ReplayPlaylistProvider` to always be an array instead of potentially undefined.
- Updated the `replays` variable in `ReplayDetailsProviders` to default to an empty array.
- Simplified the `isPending` logic in the `Playlist` component to always return false, ensuring consistent behavior.

jerryzhou196 committed:
Merge branch 'master' into replay-playlist-tab-view
